src/tools.tsx

```diff
@@ -37,15 +33,14 @@ export function withSuperJSONProps<P>(
 
 export function withSuperJSONPage<P>(
   Page: React.ComponentType<P>
-): React.ComponentType<SuperJSONProps> {
-  function WithSuperJSON(serializedProps: SuperJSONProps) {
-    if (!serializedProps._superjson) {
-      const PageWithoutProps = Page as React.ComponentType<{}>;
-      return <PageWithoutProps {...serializedProps as any as P} />;
-    }
-
-    const props = SuperJSON.deserialize<P>(serializedProps);
-    return <Page {...props} />;
+): React.ComponentType<SuperJSONProps<P>> {
+  function WithSuperJSON(serializedProps: SuperJSONProps<P>) {
+    const { _superjson, ...props } = serializedProps;
+    return (
+      <Page
+        {...SuperJSON.deserialize({ json: props as any, meta: _superjson })}
```

Comment:
This is equivalent to the before, because `deserialize` is a no-op if `meta` is undefined.